FileManager/CommandHandlers/DirectoryCommandHandler.cs

```diff
@@ -0,0 +1,407 @@
+    }
+
+    private void DeleteCatalogue(string sourceDir)
+    {
+        var dir = new DirectoryInfo(sourceDir);
+        if (!dir.Exists)
+            throw new DirectoryNotFoundException($"Исходный каталог не найден: {dir.FullName}");
+
+        DirectoryInfo[] dirs = dir.GetDirectories();
+
+        // Каждый файл в текущем каталоге удаляется
+        foreach (FileInfo file in dir.GetFiles())
+        {
+            file.Delete();
+        }
+
+        //На каждую папку в текущем каталоге вызывается этот же метод рекурсивно
+        foreach (DirectoryInfo subDir in dirs)
+        {
+            DeleteCatalogue(subDir.FullName);
+        }
+
+        // После удаления предыдущих каталогов и файлов (если есть), удаляется текущий каталог
+        Directory.Delete(sourceDir);
+    }
+
+    public override void Delete()
+    {
+        _UserInterface.Write($"Вы уверены, что хотите удалить следующий каталог и всё его содержимое:\n{_Directory.FullName}?");
+        if (_UserInterface.ReadLine("Введите \"да\" для продолжения удаления объекта > ") == "да")
+        {
+            DeleteCatalogue(_Directory.FullName);
+        }
+        else
+        {
+            _UserInterface.Write("Удаление отменено.");
+        }
+    }
+
+    private long CalcSize(DirectoryInfo dir)
+    {
+        if (!dir.Exists)
+            throw new DirectoryNotFoundException($"Исходный каталог не найден: {dir.FullName}");
+        DirectoryInfo[] dirs = dir.GetDirectories();
+        long size = 0;
+        foreach (FileInfo file in dir.GetFiles())
+        {
+            size += file.Length;
+        }
+        foreach (DirectoryInfo subDir in dirs)
+        {
+            size += CalcSize(subDir);
+        }
+        return size;
+    }
+
+    public override void Info()
+    {
+        long len = CalcSize(_Directory);
+        _UserInterface.Write($"Сведения о каталоге {_Directory.Name}.\nРазмер: {len}\nАтрибуты: {_Directory.Attributes}");
+    }
+
+    public override void PrintStatistics()
+    {
+        throw new Exception("Указана директория, а необходим файл.");
+    }
+
+    public override void Relocate(string path)
+    {
+        Copy(path);
+        DeleteCatalogue(_Directory.FullName);
+    }
+
+    public override void Rename(string name)
+    {
+        Directory.Move(_Directory.FullName, $"{Path.GetDirectoryName(_Directory.FullName)}\\{name}");
+    }
+
+    public override void Create(string path, string name)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void CreateDir(string name)
+    {
+        if (Directory.Exists(Path.GetDirectoryName(name)))
+        {
+            Directory.CreateDirectory(name);
+        }
+        else
+        {
+            Directory.CreateDirectory($"{_Directory.FullName}\\{name}");
+        }
+    }
+
+    public override void CreateFile(string name)
+    {
+        if (Directory.Exists(Path.GetDirectoryName(name)))
+        {
+            File.Create(name);
```

Comment:
```suggestion
            File.Create(name).Dispose();
```
Возвращается поток созданного и открытого на запись файла. Если его не закрыть, то файл будет заблокирован пока Сборщик мусора не удалит этот поток. Когда это будет - неизвестно (не долго ждать, но...)